app/assets/javascripts/components/slide_menu.js.jsx

```diff
@@ -0,0 +1,30 @@
+var React = require('react');
+
+var Menu = React.createClass({
+
+  getInitialState: function() {
+    return {
+      visible: false
+    };
+  },
+
+  show: function() {
+    this.setState({ visible: true });
+    document.addEventListener("click", this.hide.bind(this));
```

Comment:
A better way to do this would be to have a separate component called `MenuBackground` that fires an event when it's clicked.

But in general, calling `this.setState()` isn't great. What we really want is another module, and have our React component re-render itself when that module changes. After you see what that looks like, I think you'll be convinced it's the better way to go :)